### PR TITLE
Pass the fetchOptions to writeBatch in HttpBatchStore

### PR DESCRIPTION
**Pull request by kalman**
This allows JS to make authenticated writes to Noms.

### CODE REVIEW

**Comment by cmasone-attic** [comment]
I admit that I have no idea if this is right :-) maybe @arv?

**Comment by kalman** [comment]
@arv PTAL

File: js/noms/src/http-batch-store.js [hunk 1]
```diff
@@ -89,20 +93,22 @@ function mergeOptions(baseOpts: FetchOptions, opts: FetchOptions): FetchOptions
...
+    this._writeBatchOptions = mergeOptions(writeBatchOptions, fetchOptions);
+    this._rootOptions = fetchOptions;
     this._body = new ArrayBuffer(0);
   }
 
   async readBatch(reqs: UnsentReadMap): Promise<void> {
     const hashStrs = Object.keys(reqs);
     const body = hashStrs.map(r => 'ref=' + r).join('&');
-    const opts = Object.assign(this._readBatchOptions, {body: body});
+    const opts = Object.assign({}, this._readBatchOptions, {body});
```
Comment (arv) [inline js/noms/src/http-batch-store.js]: Or use `{...this.readBacthOptions, body}`

Comment (kalman) [inline js/noms/src/http-batch-store.js]: That didn't work for some reason, babel complained.

**Comment by kalman** [comment]
> Can this be tested?

Presumably, but there aren't any tests for the read/getRoot fetch options.

> is this actually doing anythint?

Yes?

File: js/noms/src/http-batch-store.js [hunk 2]
```diff
@@ -125,7 +131,8 @@ export class Delegate {
 
   writeBatch(hints: Set<Hash>, chunkStream: ChunkStream): Promise<void> {
     return serialize(hints, chunkStream)
-      .then(body => fetchText(this._rpc.writeValue, {method: 'POST', body}))
+      .then(body => fetchText(this._rpc.writeValue,
+                              Object.assign({}, this._writeBatchOptions, {body})))
```
Comment (kalman) [inline js/noms/src/http-batch-store.js]: This does something.

File: js/noms/src/http-batch-store.js (hunk 1 above)
Comment (kalman) [inline js/noms/src/http-batch-store.js]: I figured out why babel complains. Fixed.

Comment (kalman) [inline js/noms/src/http-batch-store.js]: False alarm. Babel still complains (but flow/eslint don't, it's a babel bug).

Comment (kalman) [inline js/noms/src/http-batch-store.js]:
```
SyntaxError: /Users/kalman/src/go/src/github.com/attic-labs/noms/js/noms/src/http-batch-store.js: Unexpected token (111:18)
  109 |     const hashStrs = Object.keys(reqs);
  110 |     const body = hashStrs.map(r => 'ref=' + r).join('&');
> 111 |     const opts = {...this._readBatchOptions, ...{body}};
      |                   ^
  112 |     const {headers, buf} = await fetchUint8Array(this._rpc.getRefs, opts);
  113 | 
  114 |     const versionErr = checkVersion(headers);
```
shrug.

**Comment by arv** [comment]
I see now... LGTM

File: js/noms/src/http-batch-store.js (hunk 1 above)
Comment (arv) [inline js/noms/src/http-batch-store.js]: I assume we haven't enabled all the needed plugins...

Comment (kalman) [inline js/noms/src/http-batch-store.js]: Well `...` is supposed to work. Maybe there is some bug with `this`.

**Pull request merged**